Document/0x05b-Basic-Security_Testing.md

```diff
@@ -536,19 +572,113 @@ The output would look like this:
+
+```python
+# frida_python.py
+import frida
+
+session = frida.get_usb_device().attach('com.android.chrome')
+
+source = """
 Java.perform(function () {
     var view = Java.use("android.view.View");
     var methods = view.class.getMethods();
     for(var i = 0; i < methods.length; i++) {
         console.log(methods[i].toString());
     }
 });
+"""
+
+script = session.create_script(source)
+script.load()
+
+session.detach()
+```
+
+In this case, running the Python script (`python3 frida_python.py`) has the same result as the previous example: it will print all methods of the `android.view.View` class to the terminal. However, you might want to work with that data from Python. Using `send` instead of `console.log` will send data in JSON format from JavaScript to Python. Please read the comments in the example below:
+
+```python
+# python3 frida_python_send.py
+import frida
+
+session = frida.get_usb_device().attach('com.android.chrome')
+
+# 1. we want to store method names inside a list
+android_view_methods = []
+
+source = """
+Java.perform(function () {
+    var view = Java.use("android.view.View");
+    var methods = view.class.getMethods();
+    for(var i = 0; i < methods.length; i++) {
+        send(methods[i].toString());
+    }
+});
+"""
+
+script = session.create_script(source)
+
+# 2. this is a callback function, only method names containing "Text" will be appended to the list
+def on_message(message, data):
+    if "Text" in message['payload']:
+        android_view_methods.append(message['payload'])
+
+# 3. we tell the script to run our callback each time a message is received
+script.on('message', on_message)
+
+script.load()
+
+# 4. we do something with the collected data, in this case we just print it
+for method in android_view_methods:
+    print(method)
+
+session.detach()
+```
+
+This effectively filters the methods and prints only the ones containing the string "Text":
+
+```java
+$ python3 frida_python_send.py
+public boolean android.view.View.canResolveTextAlignment()
+public boolean android.view.View.canResolveTextDirection()
+public void android.view.View.setTextAlignment(int)
+public void android.view.View.setTextDirection(int)
+public void android.view.View.setTooltipText(java.lang.CharSequence)
+...
 ```
 
-Frida also provides bindings for various languages, including Python, C, NodeJS, and Swift.
+At the end, it is up to you to decide where would you like to work with the data. Sometimes it will be more convenient to do it from JavaScript and in other cases Python will be the best choice. Of course you can also send messages from Python to JavaScript by using `script.post`. Refer to the Frida docs for more information about [sending](https://www.frida.re/docs/messages/#sending-messages-from-a-target-process "Sending messages from a target process") and [receiving](https://www.frida.re/docs/messages/#receiving-messages-in-a-target-process "Receiving messages in a target process") messages.
```

Comment:
```suggestion
In the end, it is up to you to decide where would you like to work with the data. Sometimes it will be more convenient to do it from JavaScript and in other cases Python will be the best choice. Of course you can also send messages from Python to JavaScript by using `script.post`. Refer to the Frida docs for more information about [sending](https://www.frida.re/docs/messages/#sending-messages-from-a-target-process "Sending messages from a target process") and [receiving](https://www.frida.re/docs/messages/#receiving-messages-in-a-target-process "Receiving messages in a target process") messages.
```